plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx

```diff
@@ -0,0 +1,182 @@
+    async () =>
+      catalogApi.getEntities({
+        kind: 'Template',
+        'metadata.name': templateName,
+      }) as Promise<TemplateEntityV1alpha1[]>,
+  );
+  return { template: data?.[0], loading: !error && !data, error };
+};
+
+const OWNER_REPO_SCHEMA = {
+  $schema: 'http://json-schema.org/draft-07/schema#' as const,
+  required: ['storePath', 'owner'],
+  properties: {
+    owner: {
+      type: 'string' as const,
+      title: 'Owner',
+      description: 'Who is going to own this component',
+    },
+    storePath: {
+      format: 'GitHub user or org / Repo name',
+      type: 'string' as const,
+      title: 'Store path',
+      description: 'GitHub store path in org/repo format',
+    },
+  },
+};
+
+const REPO_FORMAT = {
+  'GitHub user or org / Repo name': /[^\/]*\/[^\/]*/,
+};
+
+export const TemplatePage = () => {
+  const errorApi = useApi(errorApiRef);
+  const catalogApi = useApi(catalogApiRef);
+  const scaffolderApi = useApi(scaffolderApiRef);
+  const { templateName } = useParams();
+  const { template, loading } = useTemplate(templateName, catalogApi);
+
+  const [formState, setFormState] = useState({});
+
+  const handleFormReset = () => setFormState({});
+  const handleChange = (e: IChangeEvent) =>
+    setFormState({ ...formState, ...e.formData });
+
+  const [jobId, setJobId] = useState<string | null>(null);
+  const handleClose = () => setJobId(null);
+
+  const handleCreate = async () => {
+    const job = await scaffolderApi.scaffold(template!, formState);
+    setJobId(job);
+  };
+
+  const [entity, setEntity] = React.useState<TemplateEntityV1alpha1 | null>(
+    null,
+  );
+
+  const handleCreateComplete = async (job: Job) => {
+    const componentYaml = job.metadata.remoteUrl?.replace(
+      /\.git$/,
+      '/blob/master/component-info.yaml',
+    );
+
+    if (!componentYaml) {
+      errorApi.post(
+        new Error(
+          `Failed to find component-info.yaml file in ${job.metadata.remoteUrl}.`,
+        ),
+      );
+      return;
+    }
+
+    const {
+      entities: [createdEntity],
+    } = await catalogApi.addLocation('github', componentYaml);
+
+    setEntity((createdEntity as any) as TemplateEntityV1alpha1);
+  };
+
+  if (!loading && !template) {
+    errorApi.post(new Error('Template was not found.'));
+    return <Navigate to={rootRoute.path} />;
+  }
+
+  if (template && !template?.spec?.schema) {
+    errorApi.post(
+      new Error(
+        'Template schema is corrupted, please check the template.yaml file.',
+      ),
+    );
+    return <Navigate to={rootRoute.path} />;
+  }
+
+  return (
+    <Page>
+      <Header
+        pageTitleOverride="Create a new component"
+        title={
+          <>
+            Create a new component <Lifecycle alpha shorthand />
+          </>
+        }
+        subtitle="Create new software components using standard templates"
+      />
+      <Content>
+        {loading && <LinearProgress data-testid="loading-progress" />}
```

Comment:
should use our shared component `<Progress />` which also incidentally also has a testid builtin